roofit/roofitcore/inc/RooWorkspace.h

```diff
@@ -113,11 +113,25 @@ class RooWorkspace : public TNamed {
   RooAbsArg* arg(const char* name) const ;
   RooAbsArg* fundArg(const char* name) const ;
   RooArgSet argSet(const char* nameList) const ;
-  TIterator* componentIterator() const { return _allOwnedNodes.createIterator() ; }
-  const RooArgSet& components() const { return _allOwnedNodes ; }
   TObject* genobj(const char* name) const ;
   TObject* obj(const char* name) const ;
 
+  inline RooAbsPdf* pdf(std::string const& name) const { return pdf(name.c_str()); }
+  inline RooAbsReal* function(std::string const& name) const { return function(name.c_str()); }
+  inline RooRealVar* var(std::string const& name) const { return var(name.c_str()); }
+  inline RooCategory* cat(std::string const& name) const { return cat(name.c_str()); }
+  inline RooAbsCategory* catfunc(std::string const& name) const { return catfunc(name.c_str()); }
+  inline RooAbsData* data(std::string const& name) const { return data(name.c_str()); }
+  inline RooAbsData* embeddedData(std::string const& name) const { return embeddedData(name.c_str()); }
+  inline RooAbsArg* arg(std::string const& name) const { return arg(name.c_str()); }
+  inline RooAbsArg* fundArg(std::string const& name) const { return fundArg(name.c_str()); }
+  inline RooArgSet argSet(std::string const& nameList) const { return argSet(nameList.c_str()); }
+  inline TObject* genobj(std::string const& name) const { return genobj(name.c_str()); }
+  inline TObject* obj(std::string const& name) const { return obj(name.c_str()); }
+
```

Comment:
Why do we need these extra functions ?

Reply:
I am referring here to `componentIterator` and `components`

Reply:
I don't know where they are used exactly, but they were there before so I would suggest to not remove them from the interface